iaf-commons/src/main/java/nl/nn/adapterframework/util/PropertyLoader.java

```diff
@@ -0,0 +1,289 @@
+	 * @return a list of string values associated with the specified key, or the default list if the resolved property is null.
+	 * If the defaults is also null, then returns an empty list.
+	 */
+	@Nonnull
+	public List<String> getListProperty(@Nonnull String key, @Nullable String defaults) {
+		String list = getResolvedProperty(key);
+		if (list != null) {
+			return StringUtil.split(list);
+		}
+		return StringUtil.split(defaults);
+	}
+
+	/**
+	 * Add property only in the local PropertyLoader!
+	 * Do not use this method and use {@link #setProperty(String, String)} if you want to set the property globally!
+	 * <p>
+	 * This method is used by {@link Properties#load(InputStream)} to add all properties found (in a file/stream)
+	 * to the underlying {@link Hashtable}.
+	 * @deprecated Use {@link #setProperty(String, String)} instead!
+	 */
+	@Deprecated
+	@Override
+	public synchronized Object put(Object key, Object value) {
+		return super.put(key, value);
+	}
+
+	// Load methods
+
+	/**
+	 * Load the contents of a properties file.
+	 * <p>Optionally, this may be a comma-separated list of files to load, e.g.
+	 * <code><pre>log4j2.properties,DeploymentSpecifics.properties</pre></code>
+	 * which will cause both files to be loaded in the listed order.
+	 * </p>
+	 */
+	protected synchronized void load(final ClassLoader classLoader, final String filename) {
+		if(StringUtils.isEmpty(filename)) {
+			throw new IllegalStateException("file to load properties from cannot be null");
+		}
+
+		try {
+			if(classLoader == null) {
+				throw new IllegalStateException("no classloader found!");
+			}
+			List<URL> resources = Collections.list(classLoader.getResources(filename));
+			if(resources.isEmpty()) {
+				if(rootPropertyFile.equals(filename)) { //The file cannot be found, abort!
+					String msg = rootPropertyFile+ " file not found, unable to initalize AppConstants";
+					LOG.error(msg);
+					throw new MissingResourceException(msg, this.getClass().getSimpleName(), rootPropertyFile);
+				}
+
+				//An additional file to load properties from cannot be found
+				LOG.debug("cannot find resource [{}] in classloader [{}] to load additional properties from, ignoring", filename, classLoader);
+			}
+
+			//We need to reverse the loading order to make sure the parent files are loaded first
+			Collections.reverse(resources);
+
+			for (URL url : resources) {
+				try(InputStream is = url.openStream(); Reader reader = StreamUtil.getCharsetDetectingInputStreamReader(is)) {
+					load(reader);
+					LOG.info("Properties loaded from url [{}]", url::toString);
+				}
+			}
+		} catch (IOException e) {
+			LOG.error("error reading properties from [{}]", rootPropertyFile, e);
+		}
+	}
+
+	//Special Getters
+
+	/**
+	 * Gets a <code>String</code> value
+	 * Uses the {@link #getResolvedProperty(String)} method.
+	 * @param key    the Key
+	 * @param dfault the default value
+	 * @return String
+	 */
+	public String getString(String key, String dfault){
```

Comment:
Can you also rename `dfault` to `default` in this class, at multiple locations?

Reply:
The default keyword may not be used as field name.

Reply:
Ah yes, of course, it is a reserved keyword. :-)